incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -31,7 +38,28 @@ var _ = Describe("HNCConfiguration", func() {
 		}).Should(Succeed())
 	})
 
-	It("should propagate objects whose types have been added to HNCConfiguration", func() {
+	It("should set mode of Roles and RoleBindings as propagate by default", func() {
+		config := getHNCConfig(ctx)
+
+		Eventually(hasTypeWithMode("rbac.authorization.k8s.io/v1", "Role", api.Propagate, config)).Should(BeTrue())
+		Eventually(hasTypeWithMode("rbac.authorization.k8s.io/v1", "RoleBinding", api.Propagate, config)).Should(BeTrue())
```

Comment:
Just curious, where are Role and RoleBinding added to the singleton?

Reply:
When there is no singleton, the reconciler will create a default one and write it to api server in getSingleton() and writeSingleton(). The default singleton has Role and Rolebindings (http://shortn/_LxQ9fWO7i6). The two methods are called in the Reconcile() method. We force the reconciliation in SetupWithManager() so if there is no singleton in the cluster, the Reconcile() method will create a default one.

Reply:
Got you thanks!